streaming/base/storage/upload.py

```diff
@@ -32,6 +37,11 @@
 }
 
 
+class Authentication(Enum):
```

Comment:
Can we rename it to `_GCSAuthentication` ? `_` meaning private class

Reply:
I think it should be kept public? If the authentication type needs to be checked by a caller, they would use self.authentication and compare to GCSAuthentication.{something} like how it's being tested.